### PR TITLE
[SEARCHEXP-1358]Handle relative path and resolve to their real path

### PR DESCRIPTION
**Pull request by GCpigsic**
## Description
Ticket: [SEARCHEXP-1358](https://skyscanner.atlassian.net/browse/SEARCHEXP-1358)
Investigation and decision log: [Decision log](https://skyscanner.atlassian.net/wiki/spaces/SEARCHEXPTRIBE/pages/613161010/Banana+webapp+able+to+import+typed+logic)
For the consumer side, it will fail to compile when importing a typed logic from the `common` package, which is imported by  [npm local import ](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#local-paths)

We discussed the proposals listed in [this decision log](https://skyscanner.atlassian.net/wiki/spaces/SEARCHEXPTRIBE/pages/613161010/Banana+webapp+able+to+import+typed+logic), and we all agree to adopt Option 3 as a medium-term option before new potential production standard carry out. That is  👇： 

- Add a logic to handle the path which is not a part of its node_modules

ℹ️  We will release a beta version first, and verify in CI that this works in other microsites.  Once everything goes well, we will release a main version

## How to solve it
- Identify local paths and resolve to their real path

## Screenshots
Compile and build successfully in Banana with the changes

### CODE REVIEW

**Review by arnaugm — DISMISSED** [review]

File: packages/react-scripts/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@skyscanner/backpack-react-scripts",
-  "version": "10.4.0",
+  "version": "10.4.1",
```
Comment (jaysonwu991) [inline packages/react-scripts/package.json:3]: This should happen after PR is merged and new version is released.

File: packages/react-scripts/backpack-addons/babelIncludePrefixes.js [hunk 2]
```diff
@@ -1,12 +1,18 @@
...
 
 const paths = require("../config/paths");
+const path = require('path');
 const appPackageJson = require(paths.appPackageJson);
 const bpkReactScriptsConfig = appPackageJson["backpack-react-scripts"] || {};
 const customModuleRegexes = bpkReactScriptsConfig.babelIncludePrefixes
-  ? bpkReactScriptsConfig.babelIncludePrefixes.map(
-      (prefix) => new RegExp(`node_modules[\\/]${prefix}`)
-    )
+  ? bpkReactScriptsConfig.babelIncludePrefixes.map(prefix => {
```
[the remark this answers is not in the document]
Comment (GCpigsic) [reply packages/react-scripts/backpack-addons/babelIncludePrefixes.js:8]: That would be a good idea, added.
but seems BRS is lacking unit tests and there is no process in the pipeline to run the unit test automatically(I couldn’t find it if I don’t miss something), in this case, seems adding unit tests for this change doesn’t make much sense. Seems the test facility in BRS is not fully equipped, what do you think 🤔️

File: packages/react-scripts/package.json (hunk 1 above)
Comment (GCpigsic) [reply packages/react-scripts/package.json]: Good to know, thanks!

**Review by runmoore — APPROVED** [review]

**Review by arnaugm — APPROVED** [review]

**Comment by arnaugm** [comment]
beta available in https://www.npmjs.com/package/@skyscanner/backpack-react-scripts/v/10.4.1-beta.1

**Pull request merged**